src/ccl_correlation.c

```diff
@@ -518,28 +525,20 @@ INPUT:  cosmology, scale factor a, beta (= growth rate / bias),
 
 Multipole function result will be in array xi
  */
-
-void ccl_correlation_multipole(ccl_cosmology *cosmo, double a, double beta,
+void ccl_correlation_multipole(ccl_cosmology *cosmo, ccl_f2d_t *psp,
+                               double a, double beta,
                                int l, int n_s, double *s, double *xi,
                                int *status) {
   int i, N_ARR;
   double *k_arr, *pk_arr, *s_arr, *xi_arr, *xi_arr0;
 
-  if (!cosmo->computed_nonlin_power) {
-    *status = CCL_ERROR_NONLIN_POWER_INIT;
-    ccl_cosmology_set_status_message(
-      cosmo,
-      "ccl_correlation.c: ccl_correlation_multipole(): non-linear power spctrum has not been computed!");
-    return;
-  }
-
   N_ARR = (int)(cosmo->spline_params.N_K_3DCOR * log10(cosmo->spline_params.K_MAX / cosmo->spline_params.K_MIN));
 
   k_arr = ccl_log_spacing(cosmo->spline_params.K_MIN, cosmo->spline_params.K_MAX, N_ARR);
   if (k_arr == NULL) {
     *status = CCL_ERROR_MEMORY;
     strcpy(cosmo->status_message,
```

Comment:
These strcpy calls should use the set status message function.

Reply:
Done. I caught a few other of these in other files while I was at it.